ssh_filter_btrbk.sh

```diff
@@ -1,9 +1,13 @@
 #!/bin/sh
 
-set -e -u
 
+# initialise and sanitise the shell execution environment
+unset -v IFS
+export LC_ALL=C
```

Comment:
Well it should be easy for such users to override that again just for `btrbk` (e.g. again via `authorized_keys`).

But apart from that I don't think it would matter... what the SSH server sets "in between" `btrbk` from the remote side and the filter script should change anything.

And what the remote side sets (in terms of character encoding) shouldn't matter either, even if it's different from our locale.<br>
Actually only with `C` we can be sure that any strange non-character the remote side sends is still matched at all (and not produce completely undefined results).

The worst that could happen is that a command is rejected. But that should only happen in one case, namely when our fixed string parts that we match (e.g. `btrfs send`) were encoded differently on the remote side than locally.<br>
For any real world locales that should only happen when one side uses something ASCII based and the other something EBCDIC based.

Anyway, we should easily spot any issues, as the command would get rejected.

Reply:
agreed, this really should work in most cases.